Add docs for cakephp/cakephp#15371 (#6964)
cakephp/docs: en/orm/database-basics.rst

        }
    }

> Implmenting the ``ColumnSchemaAwareInterface`` gives you more control over
> custom datatypes and can avoid the need to overwrite schema definitions if your
> datatype has an unambiguous SQL column definition. For example, we could have

Reviewer: ```suggestion
Implementing ``ColumnSchemaAwareInterface`` gives you more control over
custom datatypes.  This avoids overwriting schema definitions if your
datatype has an unambiguous SQL column definition. For example, we could have
```

            ];
        }

The ``$definition`` data passed to ``convertColumnDefinition()`` could contain
the following keys:

- ``length`` The length of a column.
- ``precision`` The precision of the column if any.
- ``scale`` Can be included for SQLServer connections.

> 

Reviewer: I think we should guarantee the shared keys are set and if there is no value available, it is set to null.

"could contain" is too vague for something so standard, IMO.

Author: We would need to rework the code a bit to make those guarantees.

Reviewer: Do you feel differently? I'm arguing this point because it's literally the Type definition code which we'd theoretically use for everything.

If we make it release implementation specific, it's going to be difficult.

Author: I agree with you, the code just needs to be updated for us to be able to give a list of always present keys.

Reviewer 2: Do you want me to update the PR to add the "missing" keys?

Reviewer 2: I'll just do it.
